backend/sources/google/sender/src/main/java/co/airy/core/sources/google/Sender.java

```diff
@@ -0,0 +1,109 @@
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+
+@Component
+public class Sender implements DisposableBean, ApplicationListener<ApplicationReadyEvent> {
+    private static final Logger log = AiryLoggerFactory.getLogger(Sender.class);
+    private static final String appId = "sources.google.Sender";
+
+    private final KafkaStreamsWrapper streams;
+    private final Api api;
+    private final Mapper mapper;
+
+    Sender(KafkaStreamsWrapper streams, Api api, Mapper mapper) {
+        this.streams = streams;
+        this.api = api;
+        this.mapper = mapper;
+    }
+
+    public void startStream() {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final KStream<String, Message> messageStream = builder.<String, Message>stream(new ApplicationCommunicationMessages().name())
+                .selectKey((messageId, message) -> message.getConversationId());
+
+        final KTable<String, SendMessageRequest> contextTable = messageStream
+                .groupByKey()
+                .aggregate(SendMessageRequest::new,
+                        (conversationId, message, aggregate) -> {
+                            if (SenderType.SOURCE_CONTACT.equals(message.getSenderType())) {
+                                aggregate.setSourceConversationId(message.getSenderId());
+                            }
+                            return aggregate;
+                        });
+
+        messageStream.filter((messageId, message) -> DeliveryState.PENDING.equals(message.getDeliveryState()))
+                .join(contextTable, (message, sendMessageRequest) -> {
+                    sendMessageRequest.setMessage(message);
+                    return sendMessageRequest;
+                })
+                .mapValues(this::sendMessage)
+                .to(new ApplicationCommunicationMessages().name());
+
+        streams.start(builder.build(), appId);
+    }
+
+    private Message sendMessage(SendMessageRequest sendMessageRequest) {
+        final Message message = sendMessageRequest.getMessage();
+
+        try {
+            final SendMessagePayload sendMessagePayload = mapper.fromSendMessageRequest(sendMessageRequest);
+
+            api.sendMessage(sendMessageRequest.getSourceConversationId(), sendMessagePayload);
+
+            // TODO move the change state logic to backend/avro/message
+            message.setUpdatedAt(Instant.now().toEpochMilli());
+            message.setDeliveryState(DeliveryState.DELIVERED);
+
+            return message;
+        } catch (ApiException e) {
+            log.error(String.format("Failed to send a message to Facebook \n SendMessageRequest: %s \n FB Error Message: %s \n", sendMessageRequest, e.getMessage()), e);
+        } catch (Exception e) {
+            log.error(String.format("Failed to send a message to Facebook \n SendMessageRequest: %s", sendMessageRequest), e);
+        }
+
+        // TODO move the change state logic to backend/avro/message
```

Comment:
I think I'd do already? (but I'm find we want to follow up with it. If that, I think we should have an issue though)

Reply:
True, did not have the time yesterday but will do it right after the conversation test fix